packages/typespec-ts/src/modular/serialization/buildSerializerFunction.ts

```diff
@@ -329,7 +330,7 @@ function buildModelTypeSerializer(
     options.skipDiscriminatedUnionSuffix
   )}Serializer`;
   if (options.nameOnly) {
-    return serializerFunctionName;
+    return resolveReference(refkey(type, "serializer"));
```

Comment:
why we still need resolveReference here since we already have serializerFunctionName?

Reply:
Hah, got the idea here, to help construct the import information.